modules/mobile-agents/windows/jax-rs/src/main/java/org/wso2/carbon/mdm/mobileservices/windows/common/util/ContextListener.java

```diff
@@ -0,0 +1,115 @@
+ *  KIND, either express or implied. See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package org.wso2.carbon.mdm.mobileservices.windows.common.util;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.w3c.dom.Document;
+import org.wso2.carbon.mdm.mobileservices.windows.common.Constants;
+import org.xml.sax.SAXException;
+import javax.servlet.ServletContext;
+import javax.servlet.ServletContextEvent;
+import javax.servlet.ServletContextListener;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.File;
+import java.io.IOException;
+import org.wso2.carbon.mdm.mobileservices.windows.common.beans.WindowsPluginProperties;
+
+/**
+ * This class performs one time operations.
+ */
+public class ContextListener implements ServletContextListener {
+
+	public static final int INITIAL_VALUE = 0;
+	private static Log logger = LogFactory.getLog(ContextListener.class);
+	private static final int FIRST_ITEM = 0;
+	private static final String SIGNED_CERT_CN = "signedcertCN";
+	private static final String SIGNED_CERT_NOT_BEFORE = "signedcertnotbefore";
+	private static final String SIGNED_CERT_NOT_AFTER = "signedcertnotafter";
+	private static final String MDM_PASSWORD = "mdmpassword";
+	private static final String MDM_PRIVATE_KEY_PASSWORD = "mdmprivatekeypassword";
+
+	/**
+	 * This method loads wap-provisioning file / property file, sets wap-provisioning file and
+	 * extracted properties as attributes in servlet context.
+	 *
+	 * @param servletContextEvent - Uses when servlet communicating with servlet container.
+	 */
+	@Override
+	public void contextInitialized(ServletContextEvent servletContextEvent) {
+
+		ServletContext servletContext = servletContextEvent.getServletContext();
+		File propertyFile = new File(getClass().getClassLoader().getResource(
+				Constants.CertificateEnrolment.PROPERTIES_XML).getFile());
+		DocumentBuilderFactory docBuilderFactory = DocumentBuilderFactory.newInstance();
+		DocumentBuilder docBuilder = null;
+		try {
+			docBuilder = docBuilderFactory.newDocumentBuilder();
+		} catch (ParserConfigurationException e) {
+			logger.error("XML parsing configuration exception.");
+		}
+		Document document = null;
+		try {
+			if (docBuilder != null) {
+				document = docBuilder.parse(propertyFile);
+			}
+		} catch (SAXException e) {
+			logger.error("XML Parsing Exception.");
+		} catch (IOException e) {
+			logger.error("XML property file reading exception.");
+		}
+
+		String MDMPassword = null;
+		String MDMPrivateKeyPassword = null;
+		String signedCertCommonName = null;
+		int signedCertNotBeforeDate = INITIAL_VALUE;
+		int signedCertNotAfterDate = INITIAL_VALUE;
+
+		if (document != null) {
+		   MDMPassword = document.getElementsByTagName(MDM_PASSWORD).item(FIRST_ITEM).
```

Comment:
We discussed about renaming constants such as "FIRST_ITEM", etc to have more contextual information.